worlds/sc2/rules.py

```diff
@@ -67,7 +188,11 @@ def terran_air_anti_air(self, state: CollectionState) -> bool:
             state.has(item_names.VIKING, self.player)
             or state.has_all({item_names.WRAITH, item_names.WRAITH_ADVANCED_LASER_TECHNOLOGY}, self.player)
             or state.has_all({item_names.BATTLECRUISER, item_names.BATTLECRUISER_ATX_LASER_BATTERY}, self.player)
-            or self.advanced_tactics and state.has_any({item_names.WRAITH, item_names.VALKYRIE, item_names.BATTLECRUISER}, self.player)
+            or (
+                    self.advanced_tactics
+                    and state.has_any({item_names.WRAITH, item_names.VALKYRIE, item_names.BATTLECRUISER}, self.player)
+                    and self.weapon_armor_upgrade_count(PROGRESSIVE_TERRAN_SHIP_WEAPON, state) >= 2
```

Comment:
I don't think valks in particular need significant air upgrades to be effective vs air; I'd leave them advanced tactics without, or maybe just +1 requirement.

Reply:
I've kept that simple for now as all these units suffer from relying on low base dmg attacks. This will be surely reviewed again during the roundup

Reply:
Mentioned in #181